packages/fast-layouts-react/src/utilities/breakpoint-tracker.ts

```diff
@@ -0,0 +1,142 @@
+import { canUseDOM } from "exenv-es6";
+import { throttle } from "lodash-es";
+
+export type Breakpoint = number;
+
+export type Breakpoints = Breakpoint[];
+
+export type BreakpointTrackerCallback = (breakpoint: Breakpoint) => void;
+
+export const defaultBreakpoints: Breakpoints = [0, 540, 768, 1084, 1400, 1779];
+
+/**
+ * Identifies current breakpoint based on window width
+ */
+export function identifyBreakpoint(windowWidth: number, breakpoints: Breakpoints): number {
+    for (let breakpoint: Breakpoint = breakpoints.length - 1; breakpoint >= 0; breakpoint--) {
+        if (windowWidth >= breakpoints[breakpoint]) {
+            return breakpoint;
+        }
+    }
+}
+
+export class BreakpointTracker {
+
+    /**
+     * The current breakpoint.
+     */
+    private breakpoint: number;
+
+    /**
+     * The current breakpoint.
+     */
+    private _breakpointConfig: Breakpoints = defaultBreakpoints;
```

Comment:
why the underscore?

Reply:
@nicholasrice throws an exception about naming conflicts with the getter and setter.

Reply:
renaming based on convo regarding get/set